### PR TITLE
clearer summarise() information on unspecified .groups=

### LINKED ISSUE #5574: Better message for `summarise()` after `group_by()`
`summarise()` after `group_by()` currently gives the following message:

``` r
library(tidyverse)
mpg %>%
  group_by(cyl) %>%
  summarise(avg_cty = mean(cty))
#> `summarise()` ungrouping output (override with `.groups` argument)
#> # A tibble: 4 x 2
#>     cyl avg_cty
#>   <int>   <dbl>
#> 1     4    21.0
#> 2     5    20.5
#> 3     6    16.2
#> 4     8    12.6
```

<sup>Created on 2020-10-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

In this message, it's a bit unclear who is doing the ungrouping and who is doing the overriding. With the understanding that we want to keep this message short (<80 chars), I think the following might be a more clear message:

```
`summarise()` has ungrouped output. You can override with `.groups` argument.
``` 

- What summarise did is in the past tense.
- It's more clear that who ungrouped and who can override.

Would be good to also use code styling instead of backticks for the function names, which will save a few more characters.

### PR DESCRIPTION
**Pull request by romainfrancois**
closes #5574

### CODE REVIEW

File: tests/testthat/test-summarise-errors.txt [hunk 1]
```diff
@@ -3,19 +3,19 @@ Messages about .groups=
 =======================
 
 > ignored <- tibble(x = 1, y = 2) %>% group_by(x, y) %>% summarise()
-Message: `summarise()` regrouping output by 'x' (override with `.groups` argument)
+Message: `summarise()` has regrouped output by 'x'. You can override with `.groups=` argument.
```
Comment (romainfrancois) [inline tests/testthat/test-summarise-errors.txt:6]: perhaps "grouped" instead of "regrouped" in that case, or in all cases ?

Comment (hadley) [reply tests/testthat/test-summarise-errors.txt:6]: Yeah, I think so. 

And maybe something like ``You can override with (e.g.) `.groups = "drop"`` ?

**Comment by romainfrancois** [comment]
``` r
library(dplyr, warn.conflicts = FALSE)

ignored <- tibble(x = 1, y = 2) %>% group_by(x, y) %>% summarise()
#> `summarise()` has grouped output by 'x'. You can override with (e.g.) `.groups = "drop"`.
ignored <- tibble(x = 1, y = 2) %>% group_by(x) %>% summarise()
#> `summarise()` has ungrouped output. You can override with (e.g.) `.groups = "drop"`.
ignored <- tibble(x = 1, y = 2) %>% group_by(x, y) %>% summarise(z = c(2,2))
#> `summarise()` has grouped output by 'x', 'y'. You can override with (e.g.) `.groups = "drop"`.
ignored <- tibble(x = 1, y = 2) %>% rowwise(x, y) %>% summarise()
#> `summarise()` has grouped output by 'x', 'y'. You can override with (e.g.) `.groups = "drop"`.
ignored <- tibble(x = 1, y = 2) %>% rowwise() %>% summarise()
#> `summarise()` has ungrouped output. You can override with (e.g.) `.groups = "drop"`.
```

<sup>Created on 2020-11-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

cc @mine-cetinkaya-rundel

**Comment by mine-cetinkaya-rundel** [comment]
I like the first sentence, thanks!

This contradicts @hadley's suggestion above but I prefer not giving a particular option as a suggestion since (1) the phrasing `(e.g.)` is not super clear and (2) the point of the message is that we want people to be deliberate about what's happening to groups after `summarise()`, I think... So I'd suggest for the second sentence "You can override using the `.groups` argument."

And this is purely cosmetic but is there a way to get rid of the backticks using something like `ui_code()` (or similar, I'm not sure which one is right here). This is a minor concern but I do think they make it harder to read.

**Comment by romainfrancois** [comment]
Thanks. I do agree with you, esp that if we don't add some logic this is confusing: 

```r
ignored <- tibble(x = 1, y = 2) %>% rowwise() %>% summarise()
#> `summarise()` has ungrouped output. You can override with (e.g.) `.groups = "drop"`.
```

as `"drop"` is what is chosen automatically but loudly here. 

The backticks are everywhere in the error messages etc ... so I'm not sure we can replace them with formatting, or perhaps it's a bigger project for the `abort/warn/inform` family.

**Pull request merged**